### PR TITLE
Add up github task & service

### PR DESCRIPTION
**Pull request by linonymous**
- add service for jira & gerrit

Signed-off-by: linonymous <iamlinonymous@gmail.com>

### CODE REVIEW

File: deploy/dev/main.tf [hunk 1]
```diff
@@ -183,6 +183,36 @@ resource "aws_ecs_task_definition" "insights-connector-gerrit-task" {
...
+  family = "insights-connector-github-task"
+  requires_compatibilities = ["FARGATE"]
+  network_mode = "awsvpc"
+  cpu = "256"
+  memory = "512"
+  execution_role_arn = aws_iam_role.ecs_task_execution_role.arn
+  task_role_arn = aws_iam_role.ecs_task_role.arn
+  container_definitions = jsonencode([
+    {
+      name      = "insights-connector-git"
```
Comment (lukaszgryglicki) [inline deploy/dev/main.tf:197]: `s/insights-connector-git/insights-connector-github` ?

File: deploy/dev/main.tf [hunk 2]
```diff
@@ -252,6 +282,47 @@ resource "aws_ecs_service" "git" {
 
 }
 
+resource "aws_ecs_service" "github" {
+  name            = "insights-github"
+  cluster         = aws_ecs_cluster.insights-git-cluster.id
```
Comment (lukaszgryglicki) [inline deploy/dev/main.tf:287]: `s/aws_ecs_cluster.insights-git-cluster.id/aws_ecs_cluster.insights-github-cluster.id/`?

File: deploy/dev/main.tf [hunk 3]
```diff
@@ -252,6 +282,47 @@ resource "aws_ecs_service" "git" {
...
+  network_configuration {
+    security_groups = [aws_security_group.security_group.id]
+    subnets = [aws_subnet.main.id]
+    assign_public_ip = true
+  }
+}
+
+resource "aws_ecs_service" "jira" {
+  name            = "insights-jira"
+  cluster         = aws_ecs_cluster.insights-git-cluster.id
```
Comment (lukaszgryglicki) [inline deploy/dev/main.tf:301]: `s/aws_ecs_cluster.insights-git-cluster.id/aws_ecs_cluster.insights-github-cluster.id/`?

File: deploy/dev/main.tf [hunk 4]
```diff
@@ -252,6 +282,47 @@ resource "aws_ecs_service" "git" {
...
+  network_configuration {
+    security_groups = [aws_security_group.security_group.id]
+    subnets = [aws_subnet.main.id]
+    assign_public_ip = true
+  }
+}
+
+resource "aws_ecs_service" "gerrit" {
+  name            = "insights-gerrit"
+  cluster         = aws_ecs_cluster.insights-git-cluster.id
```
Comment (lukaszgryglicki) [inline deploy/dev/main.tf:315]: `s/insights-git-cluster/insights-gerrit-cluster/`?

File: deploy/prod/main.tf [hunk 5]
```diff
@@ -152,6 +152,35 @@ resource "aws_ecs_task_definition" "insights-connector-gerrit-task" {
...
+  family = "insights-connector-github-task"
+  requires_compatibilities = ["FARGATE"]
+  network_mode = "awsvpc"
+  cpu = "256"
+  memory = "512"
+  execution_role_arn = aws_iam_role.ecs_task_execution_role.arn
+  task_role_arn = aws_iam_role.ecs_task_role.arn
+  container_definitions = jsonencode([
+    {
+      name      = "insights-connector-git"
```
Comment (lukaszgryglicki) [inline deploy/prod/main.tf:166]: `s/insights-connector-git/insights-connector-github`?

File: deploy/prod/main.tf [hunk 6]
```diff
@@ -222,7 +251,47 @@ resource "aws_ecs_service" "git" {
 
 }
 
+resource "aws_ecs_service" "github" {
+  name            = "insights-github"
+  cluster         = aws_ecs_cluster.insights-git-cluster.id
```
Comment (lukaszgryglicki) [inline deploy/prod/main.tf:256]: `s/aws_ecs_cluster.insights-git-cluster.id/aws_ecs_cluster.insights-github-cluster.id/`?

File: deploy/prod/main.tf [hunk 7]
```diff
@@ -222,7 +251,47 @@ resource "aws_ecs_service" "git" {
...
+  network_configuration {
+    security_groups = [aws_security_group.security_group.id]
+    subnets = [aws_subnet.main.id]
+    assign_public_ip = true
+  }
+}
 
+resource "aws_ecs_service" "jira" {
+  name            = "insights-jira"
+  cluster         = aws_ecs_cluster.insights-git-cluster.id
```
Comment (lukaszgryglicki) [inline deploy/prod/main.tf:270]: `/s/aws_ecs_cluster.insights-git-cluster.id/aws_ecs_cluster.insights-jira-cluster.id/`?

**Review by ajinkyan83 — APPROVED** [review]
/lgtm

File: deploy/prod/main.tf [hunk 8]
```diff
@@ -222,7 +251,47 @@ resource "aws_ecs_service" "git" {
...
+  network_configuration {
+    security_groups = [aws_security_group.security_group.id]
+    subnets = [aws_subnet.main.id]
+    assign_public_ip = true
+  }
+}
+
+resource "aws_ecs_service" "gerrit" {
+  name            = "insights-gerrit"
+  cluster         = aws_ecs_cluster.insights-git-cluster.id
```
Comment (lukaszgryglicki) [inline deploy/prod/main.tf:284]: `/s/aws_ecs_cluster.insights-git-cluster.id/aws_ecs_cluster.insights-gerrit-cluster.id/`

File: deploy/test/main.tf [hunk 9]
```diff
@@ -152,6 +152,35 @@ resource "aws_ecs_task_definition" "insights-connector-gerrit-task" {
...
+  family = "insights-connector-github-task"
+  requires_compatibilities = ["FARGATE"]
+  network_mode = "awsvpc"
+  cpu = "256"
+  memory = "512"
+  execution_role_arn = aws_iam_role.ecs_task_execution_role.arn
+  task_role_arn = aws_iam_role.ecs_task_role.arn
+  container_definitions = jsonencode([
+    {
+      name      = "insights-connector-git"
```
Comment (lukaszgryglicki) [inline deploy/test/main.tf:166]: `s/insights-connector-git/insights-connector-github/`?

File: deploy/test/main.tf [hunk 10]
```diff
@@ -222,6 +251,47 @@ resource "aws_ecs_service" "git" {
 
 }
 
+resource "aws_ecs_service" "github" {
+  name            = "insights-github"
+  cluster         = aws_ecs_cluster.insights-git-cluster.id
```
Comment (lukaszgryglicki) [inline deploy/test/main.tf:256]: `s/aws_ecs_cluster.insights-git-cluster.id/aws_ecs_cluster.insights-github-cluster.id/`

File: deploy/test/main.tf [hunk 11]
```diff
@@ -222,6 +251,47 @@ resource "aws_ecs_service" "git" {
...
+  network_configuration {
+    security_groups = [aws_security_group.security_group.id]
+    subnets = [aws_subnet.main.id]
+    assign_public_ip = true
+  }
+}
+
+resource "aws_ecs_service" "jira" {
+  name            = "insights-jira"
+  cluster         = aws_ecs_cluster.insights-git-cluster.id
```
Comment (lukaszgryglicki) [inline deploy/test/main.tf:270]: `s/aws_ecs_cluster.insights-git-cluster.id/aws_ecs_cluster.insights-jira-cluster.id/`?

**Review by lukaszgryglicki — CHANGES_REQUESTED** [review]
Maybe some of my comments are not OK - I cannot test this - but in case if `git` cluster is shared for all other DSes then we should rename it.
Basically, I think if we define a resource for XYZ - its names should either contain XYZ or reference some shared resource - but never have `git` in their names.

File: deploy/dev/main.tf (hunk 1 above)
Comment (linonymous) [reply deploy/dev/main.tf]: updated.

File: deploy/dev/main.tf (hunk 2 above)
Comment (linonymous) [reply deploy/dev/main.tf]: renamed the cluster to a common for all name.

File: deploy/dev/main.tf (hunk 3 above)
Comment (linonymous) [reply deploy/dev/main.tf]: renamed the cluster to a common for all name.

File: deploy/prod/main.tf (hunk 5 above)
Comment (linonymous) [reply deploy/prod/main.tf]: updated

File: deploy/dev/main.tf (hunk 4 above)
Comment (linonymous) [reply deploy/dev/main.tf]: renamed the cluster to a common for all name.

File: deploy/prod/main.tf (hunk 6 above)
Comment (linonymous) [reply deploy/prod/main.tf]: renamed the cluster to a common for all name.

File: deploy/prod/main.tf (hunk 8 above)
Comment (linonymous) [reply deploy/prod/main.tf]: renamed the cluster to a common for all name.

File: deploy/prod/main.tf (hunk 7 above)
Comment (linonymous) [reply deploy/prod/main.tf]: renamed the cluster to a common for all name.

File: deploy/test/main.tf (hunk 9 above)
Comment (linonymous) [reply deploy/test/main.tf]: updated.

File: deploy/test/main.tf (hunk 10 above)
Comment (linonymous) [reply deploy/test/main.tf]: renamed the cluster to a common for all name.

File: deploy/test/main.tf (hunk 11 above)
Comment (linonymous) [reply deploy/test/main.tf]: renamed the cluster to a common for all name.

**Comment by lukaszgryglicki** [comment]
Right any changes needed on the `github` and `git` connector repos due to that? Just wanted to confirm @linonymous ?

**Review by lukaszgryglicki — APPROVED** [review]
/lgtm (there are some conflicts but this is minor).

**Comment by linonymous** [comment]
> Right any changes needed on the `github` and `git` connector repos due to that? Just wanted to confirm @linonymous ?

Yep, this ssm variable needs changes https://github.com/LF-Engineering/lfx-insights/blob/342694aaf5ca1f2e1a73b72c68fcdf720e1b377b/serverless.yml#L109

**Comment by lukaszgryglicki** [comment]
Hmm, but this is in `lfx-insights` not in the connectors themselves.

**Comment by linonymous** [comment]
> > Right any changes needed on the `github` and `git` connector repos due to that? Just wanted to confirm @linonymous ?
> 
> Yep, this ssm variable needs changes https://github.com/LF-Engineering/lfx-insights/blob/342694aaf5ca1f2e1a73b72c68fcdf720e1b377b/serverless.yml#L109 & needs redeployment.

Just checked, we dont need any changes as the existing name of the cluster is already "insights-ecs-cluster". So everything should work fine.

**Comment by linonymous** [comment]
> Hmm, but this is in `lfx-insights` not in the connectors themselves.

yep, no changes at connector end.

**Pull request merged**